llvm/lib/Target/EraVM/EraVMTieSelectOperands.cpp

```diff
@@ -58,55 +64,42 @@ char EraVMTieSelectOperands::ID = 0;
 INITIALIZE_PASS(EraVMTieSelectOperands, DEBUG_TYPE,
                 ERAVM_TIE_SELECT_OPERANDS_NAME, false, false)
 
-/// try to emit a new select instruction which contains an implicit tie
-/// so that the register allocator can coalesce both registers.
-/// Return true if a new instruction was emitted.
+bool EraVMTieSelectOperands::canPlaceTie(MachineInstr &MI,
+                                         EraVM::ArgumentKind Arg) const {
+  return MI.getOpcode() == EraVM::SELrrr ||
```

Comment:
Nit: we can also add a check that output is not tied yet as early return to this function.

Reply:
If we do that before the opcode check, then we would assume that all instructions have at least one operand and the 0 operand is a definition. This assumption can be wrong for common opcodes like KILL, and we would hit an assert.